### ISSUE TITLE
A better search integration

### ISSUE THREAD

**Issue by TROMsite** [standardnotes/desktop#734]
At the moment search is very rudimentary. I have many notes and if I search for a keyword then it filters out the notes that contain that keyword, but there is no way to jump to the part where the keyword is located inside the note. Also, I cannot search inside a single note. This to me is a big disadvantage since it is a very common feature that is used by so many. Yet I am unable to do a simple search inside a note.

I also miss the search and replace feature that many editors have.

I've read all issues here related to the search, but most are closed already with no exact resolution it seems. So I wanted to check again if you wonderful people plan to improve the search.

As I understand SN uses external editors, but the Bold Editor is built by you. So adding a search function across so many editors seems not possible, although am wondering don't they provide an API that an be used for such tasks? Do you plan to improve the search for at least the editors that you maintain?

Basically a good search would be to:
1. Search through all notes and jump to the lines in the notes where the keyword is found. Similar to how PDF viewers work.
2. Ability to search directly inside a note and jump to the link containing the keyword.
3. Ability to search and replace.

Cheers!
